client/src/app/add/add-note.component.ts

```diff
@@ -0,0 +1,62 @@
+import { Component, OnInit } from '@angular/core';
+import { FormBuilder, FormControl, FormGroup, Validators } from '@angular/forms';
+import { MatSnackBar } from '@angular/material/snack-bar';
+import { Router } from '@angular/router';
+import { Note } from '../note';
+import { NotesService } from '../notes.service';
+
+@Component({
+  selector: 'app-add-note',
+  templateUrl: './add-note.component.html',
+  styleUrls: ['./add-note.component.scss']
+})
+export class AddNoteComponent implements OnInit {
+
+  addNoteForm: FormGroup;
+
+  note: Note;
+
+  constructor(private fb: FormBuilder, private noteService: NotesService, private snackBar: MatSnackBar, private router: Router) {
+  }
+
+  // not sure if this name is magical and making it be found or if I'm missing something,
+  // but this is where the red text that shows up (when there is invalid input) comes from
+  add_note_validation_messages = {
+    body: [
+      {type: 'required', message: 'Body is required'},
+      {type: 'minlength', message: 'Body must be at least 2 characters long'},
+      {type: 'maxlength', message: 'Body cannot be more than 50 characters long'}
+    ]
+  };
+
+  createForms() {
+
+    // add note form validations
+    this.addNoteForm = this.fb.group({
+      body: new FormControl('', Validators.compose([
+        Validators.required,
+        Validators.minLength(2),
```

Comment:
Do we need a minimum length at all? I mean, I don't think it's *harming* anything here, but why do we care if the message is one character long?

Reply:
I don't see any situation where the user will be posting a note that's 1 character long.

Reply:
I'd like to formally disagree with this decision. However, I'm willing to let it stand.